.github/workflows/main.yml

```diff
@@ -0,0 +1,85 @@
+name: Main workflow
+
+on:
+  push:
+    branches:
+    - master
+  pull_request:
+
+jobs:
+  build_lint_and_test:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v1
+
+      # - name: Install dependencies
+      #   uses: bahmutov/npm-install@v1
+      - name: Setup Node 12.x
+        uses: actions/setup-node@v1
+        with:
+          node-version: '12.x'
+      - name: Get yarn cache
+        id: yarn-cache
+        run: echo "::set-output name=dir::$(yarn cache dir)"
+      - uses: actions/cache@v1
+        with:
+          path: ${{ steps.yarn-cache.outputs.dir }}
+          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-yarn-
+      - name: Install dependencies
+        run: yarn install --frozen-lockfile
+
+      - name: Lint text files
+        run: yarn run jest --projects jest.{eslint,stylelint,test,text}.config.js --maxWorkers=3 --reporters jest-silent-reporter
+
+      - name: Building websites for production
+        run: yarn build
+
+      - name: Running Visual Regression Tests
+        uses: percy/snapshot-action@v0.1.1
+        with:
+          build-directory: public
+        env:
+          PERCY_TOKEN: ${{ secrets.PERCY_TOKEN_SMOKE_TESTS }}
+
+      - name: Running End-to-End tests
+        uses: cypress-io/github-action@v1.16.1
+        with:
+          headless: true
+          start: 'yarn run serve -l 8000 -n public'
+          wait-on: 'http://localhost:8000/docs-smoke-test'
+          env: 'CI=true'
+          command-prefix: 'yarn run percy exec --'
```

Comment:
FYI: this option does not exist yet. I opened a PR to add support for it: https://github.com/cypress-io/github-action/pull/61

Without that, we can't run Percy with Cypress.

On the other hand, the GitHub Action of Percy does not support waiting for a server to run, therefore we can't use that either. https://github.com/marketplace/actions/percy

Reply:
YAY! https://github.com/cypress-io/github-action/releases/tag/v1.20.0